File: cpp/SenjataDagangan.cpp
#include "BarangDagangan.cpp"

// Kelas cucu: mewarisi Senjata + BarangDagangan, menambah kode lisensi & penjual
class SenjataDagangan : public BarangDagangan {

private:
    string kodeLisensiJual; // Kode lisensi jual
    string penjual;         // Nama toko/penjual

public:
    // Constructor kosong
    SenjataDagangan() {}

    // Constructor dengan Parameter
    SenjataDagangan(string id, string n, string j, string s, string m, double h, int st, string k, bool g, string kode, string p) {
        setIdSenjata(id);
        setNama(n);
        setJenis(j);
        setSpesifikasi(s);
        setMadeIn(m);
        setHarga(h);
        setStok(st);
        setKondisi(k);
        setGaransi(g);
        kodeLisensiJual = kode;
        penjual = p;
    }

    // Getter Method
    string getKodeLisensiJual()const{
        return kodeLisensiJual;
    }
    string getPenjual()const{
        return penjual;
    }

    // Setter Method
    void setKodeLisensiJual(string kode){
        kodeLisensiJual = kode;
    }
    void setPenjual(string p){
        penjual = p;
    }

    // Mengubah satu objek menjadi vector<string> untuk tabel dinamis (11 kolom)
    vector<string> toRow() const {
        string garansiStr = getGaransi() ? "Ya" : "Tidak";

        // Format harga menjadi Rp 12.500.000 (titik ribuan)
        long long h   = (long long) getHarga();
        string hargaStr = to_string(h);
        string formatted;
        int count = 0;

        for (int i = hargaStr.size() - 1; i >= 0; i--) {
            formatted.push_back(hargaStr[i]);
            count++;
            if (count == 3 && i != 0) {
                formatted.push_back('.');
                count = 0;
            }
        }
        reverse(formatted.begin(), formatted.end());
        formatted = "Rp " + formatted;

        return {
            getIdSenjata(),
            getNama(),
            getJenis(),
            getSpesifikasi(),
            getMadeIn(),
            formatted,
            to_string(getStok()),
            getKondisi(),
            garansiStr,
            kodeLisensiJual,
            penjual
        };
    }

    // Destructor
    ~SenjataDagangan() {}
};

File: php/SenjataDagangan.php
<?php
require_once "BarangDagangan.php";

// Kelas cucu: mewarisi Senjata + BarangDagangan, menambah lisensi, penjual, dan foto_produk (khusus PHP)
class SenjataDagangan extends BarangDagangan {
    private $kodeLisensiJual; // Kode lisensi jual
    private $penjual;         // Nama toko/penjual
    private $foto_produk;     // Path lokal gambar (KHUSUS PHP, tidak ada di cpp/java/python)

    // Constructor lengkap 12 atribut (5 induk + 4 dagang + 3 cucu)
    public function __construct($idSenjata = "", $nama = "", $jenis = "", $spesifikasi = "", $madeIn = "",
                                $harga = 0, $stok = 0, $kondisi = "", $garansi = false,
                                $kodeLisensiJual = "", $penjual = "", $foto_produk = "") {
        parent::__construct($idSenjata, $nama, $jenis, $spesifikasi, $madeIn, $harga, $stok, $kondisi, $garansi);
        $this->kodeLisensiJual = $kodeLisensiJual;
        $this->penjual = $penjual;
        $this->foto_produk = $foto_produk;
    }

    // Getter Method
    public function getKodeLisensiJual() { return $this->kodeLisensiJual; }
    public function getPenjual() { return $this->penjual; }
    public function getFotoProduk() { return $this->foto_produk; }

    // Setter Method
    public function setKodeLisensiJual($kode) { $this->kodeLisensiJual = $kode; }
    public function setPenjual($penjual) { $this->penjual = $penjual; }
    public function setFotoProduk($foto) { $this->foto_produk = $foto; }

    // Helper: format harga menjadi Rp 12.500.000
    public function getHargaFormatted() {
        return "Rp " . number_format($this->getHarga(), 0, ',', '.');
    }

    // Helper: ubah boolean garansi menjadi Ya/Tidak
    public function getGaransiStr() {
        return $this->getGaransi() ? "Ya" : "Tidak";
    }
}
?>
